Source/WebKit/GPUProcess/media/RemoteMediaPlayerProxy.cpp

```diff
@@ -1157,12 +1157,14 @@ void RemoteMediaPlayerProxy::performTaskAtTime(const MediaTime& taskTime, Perfor
     }
 
     m_performTaskAtTimeCompletionHandler = WTFMove(completionHandler);
-    player->performTaskAtTime([this, weakThis = WeakPtr { *this }]() mutable {
-        if (!weakThis || !m_performTaskAtTimeCompletionHandler)
-            return;
+    player->performTaskAtTime([weakThis = WeakPtr { *this }]() mutable {
+        if (RefPtr protectedThis = weakThis.get()) {
+            if (!protectedThis->m_performTaskAtTimeCompletionHandler)
+                return;
 
-        auto completionHandler = WTFMove(m_performTaskAtTimeCompletionHandler);
-        completionHandler(protectedPlayer()->currentTime());
+            auto completionHandler = WTFMove(protectedThis->m_performTaskAtTimeCompletionHandler);
```

Comment:
nit: Ideally this would use `std::exchange(protectedThis->m_performTaskAtTimeCompletionHandler, nullptr)` instead of `WTFMove()` to be 100% sure we're leaving the data member in a good state in case it gets re-used later.

Reply:
Yes, that sounds safer indeed, done.